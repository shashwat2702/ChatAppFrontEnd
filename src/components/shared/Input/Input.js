import React from 'react'
import './Input.scss';

export default function Input(props) {
  return (
    <div className="inputComponent">
        <label htmlFor={props.id}>
            {props.label}
            <span className={props.className}>
                {props.spanContent}
            </span>
        </label>
        <input
            type={props.type}
            onBlur={props.onBlur}
            onChange={props.onChange}
            placeholder={props.placeholder}
            value={props.value}
            id={props.id}
            autoComplete={props.autoComplete}
        />
    </div>
  )
}
