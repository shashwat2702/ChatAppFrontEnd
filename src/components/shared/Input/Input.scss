$body-bg: #c1bdba;
$form-bg: #13232f;
$white: #ffffff;

$main: #1ab188;
$main-light: lighten($main,5%);
$main-dark: darken($main,5%);

$gray-light: #a0b3b0;
$gray: #ddd;

$thin: 300;
$normal: 400;
$bold: 600;
$br: 4px;
.inputComponent{
    label {
        left:13px;
        color:rgba($white,1);
        font-size:22px;
      }
      input {
        font-size:22px;
        display:block;
        width:100%;
        height:100%;
        padding:5px 10px;
        background:none;
        background-image:none;
        border:1px solid $gray-light;
        color:$white;
        border-radius:0;
        transition:border-color .25s ease, box-shadow .25s ease;
        &:focus {
          outline:0;
          border-color:$main;
        }
      }
}