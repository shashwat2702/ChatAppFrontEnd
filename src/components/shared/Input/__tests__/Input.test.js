import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input';

jest.mock('react-router-dom');

describe('Input Component snapshot', () => {
  const props = {
    id: 'inputField',
    label: 'Name',
    className: 'spanStyle',
    spanContent: '*',
    type: 'text',
    autoComplete: 'off',
  };
  it('should render correctly', () => {
    const HeaderSnap = renderer.create(<Input {...props} />);
    const HeaderJson = HeaderSnap.toJSON();
    expect(HeaderJson).toMatchSnapshot();
  });
});
