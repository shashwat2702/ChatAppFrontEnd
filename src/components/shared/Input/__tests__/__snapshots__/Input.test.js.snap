// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component snapshot should render correctly 1`] = `
<div
  className="inputComponent"
>
  <label
    htmlFor="inputField"
  >
    Name
    <span
      className="spanStyle"
    >
      *
    </span>
  </label>
  <input
    autoComplete="off"
    id="inputField"
    type="text"
  />
</div>
`;
